build/openwisp_base/Dockerfile

```diff
@@ -20,21 +20,24 @@ RUN apt update && \
 
 ENV PYTHONPATH=/install/lib/python3.7/site-packages
 # Python Packages
-RUN pip install https://github.com/openwisp/openwisp-controller/tarball/74600dc85f3ff0d4383c96c54bc00db122292a22 \
-                --upgrade --prefix='/install'
-RUN pip install https://github.com/openwisp/openwisp-network-topology/tarball/417ebec33ee38885501d0915cb7e37a415ed413d \
-                --upgrade --prefix='/install'
-RUN pip install https://github.com/openwisp/django-freeradius/tarball/4fbf6ca98cea726d22dae54a0684c9468fc752a1 \
-                --upgrade --prefix='/install'
-RUN pip install https://github.com/openwisp/openwisp-radius/tarball/af89345626076e27c81168c20daaa98f922223a0 \
-                --upgrade --prefix='/install'
-RUN pip install https://github.com/openwisp/django-netjsonconfig/tarball/ce0a694f1917f8c65baf0b88bb4f8ed014a577a9 \
-                --upgrade --prefix='/install'
-RUN pip install https://github.com/openwisp/django-netjsongraph/tarball/e18e5245724e58f0ef5310f0c57ba5347b8e32e7 \
-                --upgrade --prefix='/install'
-RUN pip install --prefix='/install' --upgrade openwisp-users~=0.1
-RUN pip install --prefix='/install' --upgrade django-x509~=0.5
-RUN pip install --prefix='/install' --upgrade openwisp-utils~=0.3
+ARG OPENWISP_CONTROLLER_SOURCE=openwisp-controller
+RUN pip install --upgrade --prefix='/install' ${OPENWISP_CONTROLLER_SOURCE}
+ARG OPENWISP_TOPOLOGY_SOURCE=openwisp-network-topology
+RUN pip install --upgrade --prefix='/install' ${OPENWISP_TOPOLOGY_SOURCE}
+ARG DJANGO_FREERADIUS_SOURCE=django-freeradius
+RUN pip install --upgrade --prefix='/install' ${DJANGO_FREERADIUS_SOURCE}
+ARG OPENWISP_RADIUS_SOURCE=openwisp-radius
+RUN pip install --upgrade --prefix='/install' ${OPENWISP_RADIUS_SOURCE}
+ARG OPENWISP_USERS_SOURCE=openwisp-users
+RUN pip install --upgrade --prefix='/install' ${OPENWISP_USERS_SOURCE}
+ARG DJANGO_NETJSONCONFIG_SOURCE=django-netjsonconfig
+RUN pip install --upgrade --prefix='/install' ${DJANGO_NETJSONCONFIG_SOURCE}
+ARG DJANGO_NETJSONGRAPH_SOURCE=django-netjsongraph
+RUN pip install --upgrade --prefix='/install' ${DJANGO_NETJSONGRAPH_SOURCE}
+ARG DJANGO_X509_SOURCE=django-x509
+RUN pip install --upgrade --prefix='/install' ${DJANGO_X509_SOURCE}
+ARG OPENWISP_UTILS_SOURCE=openwisp-utils
+RUN pip install --upgrade --prefix='/install' ${OPENWISP_UTILS_SOURCE}
 RUN pip install --prefix='/install' asgi_redis service_identity django-redis psycopg2 uwsgi sentry-sdk \
                                     supervisor celery django-celery-beat redis==2.10.6
```

Comment:
@atb00ker do we still need do pin down `redis==2.10.6` or we can unpin it?

Reply:
@nemesisdesign yes, I think this can be removed when daphne is updated, which will happen when `openwisp-controller` moves to `channels2`! 
@marfgold1 Can you double check it? Thanks! :smile:

Reply:
It's still working even if it's removed :+1: